internal/oci/runtime_vm.go

```diff
@@ -52,6 +53,33 @@ type containerInfo struct {
 	cio *cio.ContainerIO
 }
 
+func buildShimV2Path(path string) string {
```

Comment:
I think this may be better suited for config validation step in pkg/config/config.go

that way, we fail early if a user configured vm runtime incorrectly

Reply:
@haircommander, let me see if I understand your suggestion correctly.  Your suggestion seems to be along the lines of what @fgiudici suggested here: https://github.com/cri-o/cri-o/pull/4590#discussion_r580187812, and I'm totally in for that.

With that in mind, we'd still need to build the "ShimV2Path", as what we pass to containerd is **not** exactly the same as the path of the binary.

For instance, for let's take the `containerd-shim-kata-v2` binary as example:
```
containerd-shim-kata-v2
                ----
                  |
                  v
         this should not change, and
    every other "-" should be replaces by "."
```
This will lead us to "containerd.shim.kata.v2" as the name passed to `containerd`.


Now, let's take `containerd-shim-kata-qemu-v2`, binary as example:
```
containerd-shim-kata-qemu-v2
                ---------
                    |
                    v
             this should not change, and
        every other "-" should be replaces by "."
```
This will lead us to "containerd.shim.kata-qemu.v2" as the name passed to `contained`.

So, AFAIU, we could validate that the binary name would be something like:
`(/\\w\/-+)+/containerd-shim-\\w+-v2` as part of the pkg/config/config.go (and, again, that does make sense), but we would still have to convert the `containerd-shim-$something-v2" to `containerd.shim.$something.v2`.

Based on the explanation above, do you still think the whole `buildShimV2Path` could be done at the config validation step in `pkg/config/config.go`?